tx.go

```diff
@@ -142,29 +105,69 @@ func (tx *Tx) ReadFrom(r io.Reader) (int64, error) {
 
 	tx.Version = binary.LittleEndian.Uint32(version)
 
+	extended := false
+
 	var inputCount VarInt
-	n64, err := inputCount.ReadFrom(r)
+
+	n64, err = inputCount.ReadFrom(r)
 	bytesRead += n64
 	if err != nil {
 		return bytesRead, err
 	}
 
+	var outputCount VarInt
+	locktime := make([]byte, 4)
+
+	if inputCount == 0 {
+		// The next bytes are either 0xEF or a varint that specifies the number of outputs
+		// Read 1 more byte to see if this is in extended format...
+		n64, err = outputCount.ReadFrom(r)
+		bytesRead += n64
+		if err != nil {
+			return bytesRead, err
+		}
+
+		if outputCount == 0 {
```

Comment:
This nesting could be removed with 
```
if outputCount > 0 {
   return bytesRead, nil
}

... additional logic
```

Reply:
Returning here is different from the current behaviour.  I have added a comment to try to explain what's going on.